### PR TITLE
Use vanilla syntax for commands usage

### PR DESCRIPTION
**Pull request by imDaniX**
The reason why I made it is simple - it adds some more consistency with the vanilla commands syntax and resolves situations where usage is not obvious, like for `sethome` command.

Anyway, I'm not sure if this PR is actually required. It adds new element - parentheses - which slightly increases overall complexity.
As for now, leaving this PR as a draft for further conversation, also it needs messages.yml update.
If this idea will be declined, I will commit another PR with fixing inconsistencies(like `[player]`-`[playername]`) and syntax typos(like `afk`'s one).

### CODE REVIEW

**Comment by pop4959** [comment]
Is there somewhere where the vanilla syntax requirements are specified in detail? Currently Essentials (and most other plugins) use a common format recognized by most developers using the Unix/Linux command line. I only worry that changing the format here will make it more confusing for contributors adding or changing commands. Aside from that, sounds like a decent suggestion.

Keep in mind that this doesn't mean the idea is accepted, I think it would be good for the other team members to voice their concerns first if they have any.

**Comment by imDaniX** [comment]
@pop4959 Well, there's a [small note](https://minecraft.gamepedia.com/Commands#Syntax) on Minecraft's gamepedia, which actually describes everything. I think it's not so hard to get into, because the only major thing to remember - \<argument> should be used when you want to get _custom_ input from the player.

**Comment by Andre601** [comment]
While I don't mind this style change do I find it a bit confusing how the syntax now looks.

According to the page you linked is `[text]` used to indicate optional and `<text>` to indicate required options, which I also learned like that.
The issue I now have is, that you use `[<text>]` at various places which according to the above logic makes 0 sense. This text essentially tells me that the argument is optional, but then is also required?

The syntax could just stay as `/<command> [player]` and be fine for all I know as it indicates an optional Player parameter you can set.
The `<>` is only useful in cases where you either have multiple options for that argument, separated by `|` or if depending on the argument, the following syntax afterwards changes (Depends on what type you defined there. Hope this makes sense.)

One noteworthy change to make:
The syntax for afk is `/<command> [<player>] [<message...>]` which A) doesn't make sense as described earlier with the conflicting argument types and B) is to my knowledge NOT how the AFK command works.
It is either `/afk Andre_601` to set a Player AFK or `/afk I'm off to sleep` to set yourself AFK with a reason. So the syntax should actually be `/<command> [<player>|<message...>]` to indicate either option (The `<>` is used here now due to the `|` to indicate that those aren't literal options (Not actually "player" and "message..." but placeholders for what you put there)

If the `/afk Andre_601` command actually allows setting a optional reason for that player would the syntax be `/<command> [<player> [message...]|<message...>]`

Here's also a picture of how I use this syntax in my Bot's help command when depicting the commands.
You can clearly recognize what stuff is optional, which are literal and which are arguments and also required stuff.

image

File: Essentials/src/main/resources/plugin.yml [hunk 1]
```diff
@@ -79,15 +79,15 @@ commands:
...
     description: Clear all items in your inventory.
-    usage: /<command> [player|*] [item[:<data>]|*|**] [amount]
+    usage: /<command> [(<player>|*)] [(<item>|*|**) [<amount>]]
     aliases: [ci,eci,clean,eclean,clear,eclear,clearinvent,eclearinvent,eclearinventory]
   clearinventoryconfirmtoggle:
     description: Toggles whether you are prompted to confirm inventory clears.
     usage: /<command>
     aliases: [eclearinventoryconfirmtoggle, clearinventoryconfirmoff, eclearinventoryconfirmoff, clearconfirmoff, eclearconfirmoff, clearconfirmon, eclearconfirmon, clearconfirm, eclearconfirm]
   condense:
     description: Condenses items into a more compact blocks.
```
Comment (Andre601) [inline Essentials/src/main/resources/plugin.yml:89]: While you're at it changing the plugin.yml could you also correct this typo here?
I'm sure the EssX team would be happy about it.

**Comment by imDaniX** [comment]
@Andre601
> According to the page you linked is [text] used to indicate optional and <text> to indicate required options, which I also learned like that.
> The issue I now have is, that you use [<text>] at various places which according to the above logic makes 0 sense. This text essentially tells me that the argument is optional, but then is also required?

It actually makes sense. As I said before, the main difference is that `<key>` means that user have to type something there, and everything else should be typed as-is. In that case, `/test [world]` and `/test [<world>]` are different commands - first one asks player to write exactly `/test world`, while the second one asks to write world name, e.g. `/test world_the_end`. Also, it's actually used in vanilla, like, everywhere (for example [ban](https://minecraft.gamepedia.com/Commands/ban), [attribute](https://minecraft.gamepedia.com/Commands/attribute), just check the wiki).

> One noteworthy change to make:
> The syntax for afk is /<command> [<player>] [<message...>] which A) doesn't make sense as described earlier with the conflicting argument types and B) is to my knowledge NOT how the AFK command works.

Well, a) does make sense as described earlier and b) according to [source code](https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/main/java/com/earth2me/essentials/commands/Commandafk.java#L26), that's how /afk works.

> If the /afk Andre_601 command actually allows setting a optional reason for that player would the syntax be /<command> [<player> [message...]|<message...>]

Why? These arguments are completely optional - you can either ignore `[<player>]`, `[<message...>]` or both.

Anyway, I think this _draft_ PR will eventually be closed due to #4057, and I think it's actually for the better.

**Comment by imDaniX** [comment]
o7

**Comment by mdcfe** [comment]
Argh I had this PR here to leave a comment and forgot...

But yes, this is largely superseded by #4057, which retains the Bukkit/EssentialsX syntax but adds multiple usage messages.

**Comment by pop4959** [comment]
Just popping in briefly to say that I do think the idea of normalizing all of the usage messages is a pretty good idea. If that's already been done in #4057 though, perhaps this doesn't matter so much. The vanilla syntax is nice though, once you get used to it.

**Pull request closed without merging**